storage_service/locations/models/rclone.py

```diff
@@ -56,17 +56,21 @@ def _execute_subprocess(self, subcommand):
         """
         cmd = ["rclone"] + subcommand
         try:
-            with subprocess.Popen(
-                cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE
-            ) as proc:
-                (stdout, stderr) = proc.communicate()
+            proc = subprocess.Popen(cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+            (stdout, stderr) = proc.communicate()
 
-                LOGGER.debug("rclone cmd: %s", cmd)
-                LOGGER.debug("rclone stdout: %s", stdout)
-                if stderr:
-                    LOGGER.warning(stderr)
+            LOGGER.debug("rclone cmd: %s", cmd)
+            LOGGER.debug("rclone stdout: %s", stdout)
+            if stderr:
+                LOGGER.warning("rclone stderr: %s", stderr)
 
-                return stdout, stderr
+            if proc.returncode != 0:
```

Comment:
Should we configure a retry system for retryable errorrs (Return code 5)?

Reply:
Nice suggestion! Commit https://github.com/artefactual/archivematica-storage-service/pull/635/commits/5a17501b25d10845c6e0f655b706756d0713198c introduces up to 5 retries, with two seconds in between each